esp/esdllib/esdl_script.cpp

```diff
@@ -328,42 +666,86 @@ class CEsdlTransformOperationSetValue : public CEsdlTransformOperationBase
         }
         return m_target.str();
     }
-    virtual bool doSet(IXpathContext * xpathContext, IPropertyTree *tree, const char *value)
+    virtual bool doSet(IXpathContext * sourceContext, IXpathContext *targetContext, const char *value)
     {
         StringBuffer xpath;
-        const char *target = getTargetPath(xpathContext, xpath);
-        ensurePTree(tree, target);
-        tree->setProp(target, value);
+        const char *target = getTargetPath(sourceContext, xpath);
+        targetContext->ensureSetValue(target, value, m_required);
         return true;
     }
 };
 
-class CEsdlTransformOperationRenameNode : public CEsdlTransformOperationBase
+class CEsdlTransformOperationNamespace : public CEsdlTransformOperationWithoutChildren
+{
+protected:
+    StringAttr m_prefix;
+    StringAttr m_uri;
+    bool m_current = false;
+
+public:
+    CEsdlTransformOperationNamespace(XmlPullParser &xpp, StartTag &stag, const StringBuffer &prefix) : CEsdlTransformOperationWithoutChildren(xpp, stag, prefix)
+    {
+        if (m_traceName.isEmpty())
+            m_traceName.set(stag.getValue("name"));
+
+        const char *pfx = stag.getValue("prefix");
+        const char *uri = stag.getValue("uri");
+        if (!pfx && isEmptyString(uri))
```

Comment:
Not a big deal at all but why not use isEmptyString() to check both?

Reply:
"" is a valid prefix, but if you are relying on the uri it can't be empty.  That is , pfx=nullptr, uri="" has nothing to match on, pfx="", uri=nullprt can search for an existing prefix definition for "".